File: Server/src/main/java/jk_5/nailed/server/command/CommandFirework.java
package jk_5.nailed.server.command;

import jk_5.nailed.api.map.Map;
import jk_5.nailed.effect.FireworkRandomizer;
import net.minecraft.command.ICommandSender;
import net.minecraft.util.ChunkCoordinates;

/**
 * No description given
 *
 * @author jk-5
 */
public class CommandFirework extends NailedCommand {

    public CommandFirework(){
        super("firework");
    }

    @Override
    public void processCommandWithMap(ICommandSender sender, Map map, String[] args){
        ChunkCoordinates coords = sender.getPlayerCoordinates();
        FireworkRandomizer.getRandomEffect().toFirework().spawnInWorld(map, coords.posX, coords.posY, coords.posZ);
    }
}

File: Server/src/main/scala/jk_5/nailed/server/command/CommandFirework.scala
package jk_5.nailed.server.command

import net.minecraft.command.ICommandSender
import jk_5.nailed.api.map.Map
import jk_5.nailed.effect.FireworkRandomizer

/**
 * No description given
 *
 * @author jk-5
 */
object CommandFirework extends ScalaCommand {

  override val name = "firework"
  override val usage = "/firework - Spawns a random firework rocket"

  override def processCommandWithMap(sender: ICommandSender, map: Map, args: Array[String]){
    val coords = sender.getPlayerCoordinates
    FireworkRandomizer.getRandomEffect.toFirework.spawnInWorld(map, coords.posX, coords.posY, coords.posZ)
  }
}
